kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMSettingsConverter.java

```diff
@@ -81,6 +82,11 @@ public static Map<String, Variant<?>> buildIpv4Settings(NetworkProperties props,
             String address = props.get(String.class, "net.interface.%s.config.ip4.address", iface);
             Short prefix = props.get(Short.class, "net.interface.%s.config.ip4.prefix", iface);
 
+            if (ip4Status.equals(KuraIpStatus.ENABLEDWAN) && Objects.nonNull(address) && !address.isEmpty()) {
```

Comment:
No need to check for non null `address` and empty address. It's all taken care by the `NetworkProperties.get` method.

```suggestion
            if (ip4Status.equals(KuraIpStatus.ENABLEDWAN)) {
```